pkg/kubeclient/kubelet_client.go

```diff
@@ -33,12 +34,22 @@ type KubeHttpClientInterface interface {
 	GetMachineCpuFrequency(host string) (uint64, error)
 }
 
+// Cache structure.
+// TODO(MB): Make sure the nodes that are no longer discovered are being removed from the cache!
```

Comment:
I think this is important implementation of the entire validation flow. For now, the cache never expires. However, the node status changes are common in kubernetes so it needs to be taken care here. We can expire the cached data with the cache library already used in kubeturbo.

Reply:
We've agreed that this isn't needed for this release. Also, the cache invalidation should not be a timing based, but driven by what's being discovered.